lexos/static/js/scripts_manage.js

```diff
@@ -1,803 +1,906 @@
+function deleteAllSelected (selected_rows) {
+    row_ids = []
+    selected_rows.each(function () {
+        id = $(this).attr('id')
+        row_ids.push(id)
     })
+    html = '<p>Are you sure you wish to delete the selected documents?</p>'
+    html += '<span id="deleteIds" style="display:none;">' + row_ids.toString() + '</span>'
+    footer = '<div class="modal-footer"><button type="button" data-dismiss="modal" class="btn btn-primary" id="confirm-delete-bttn" style="margin-left:2px;margin-right:2px;">Delete</button><button type="button" data-dismiss="modal" class="btn" style="margin-left:2px;margin-right:2px;">Cancel</button></div>'
+    $('#delete-modal .modal-body').html(html)
+    $('#delete-modal .modal-body').append(footer)
+    $('#delete-modal').modal()
+        .one('click', '#confirm-delete-bttn', function () {
+            row_ids = $('#deleteIds').text()
+            deleteSelected(row_ids)
+        })
 }
+
 /* #### END OF deleteAllSelected #### */
 
 /* #### unique() #### */
+
 // Helper function ensures id lists have no duplicates
-function unique(array) {
-  return $.grep(array, function (el, index) {
-    return index === $.inArray(el, array)
-  })
+function unique (array) {
+    return $.grep(array, function (el, index) {
+        return index === $.inArray(el, array)
+    })
 }
 
 /* #### prepareContextMenu() #### */
 // Helper function to change configure the context menu based on
 // the number of rows currently selected
 
-function prepareContextMenu() {
-  // Refresh all options
-  $('#context-menu').find('li').removeClass('disabled')
-  $('#context-menu').find('li').find('a').removeProp('disabled')
-
-  // Comparison values
-  num_rows = table.rows().ids().length
-  num_rows_selected = table.rows({ selected: true }).ids().length
-
-  // Set config options -- Numbers refer to li elements, including dividers
-  switch (true) {
-    case num_rows_selected == 0: // No rows selected
-      opts = [6, 8, 9]
-      break
-    case num_rows_selected == 1: // 1 row selected
-      opts = [8, 9]
-      break
-    case num_rows_selected > 1 && num_rows_selected < num_rows: // More than 1 row selected
-      opts = []
-      break
-    case num_rows_selected == num_rows: // All rows selected
-      opts = [5]
-      break
-    default: // Just in case
-      opts = []
-  }
-
-  // Disable configured options
-  $.each(opts, function (k, opt) {
-    $('#context-menu').find('li').eq(opt).attr('class', 'disabled')
-    $('#context-menu').find('li').eq(opt).find('a').prop('disabled', true)
-  })
+function prepareContextMenu () {
+    // Refresh all options
+    $('#context-menu').find('li').removeClass('disabled')
+    $('#context-menu').find('li').find('a').removeProp('disabled')
+
+    // Comparison values
+    num_rows = table.rows().ids().length
+    num_rows_selected = table.rows({selected: true}).ids().length
+
+    // Set config options -- Numbers refer to li elements, including dividers
+    switch (true) {
```

Comment:
This is a unconventional switch statement. What is wrong with plain old `if` and `else if`